github_workflow.md

```diff
@@ -0,0 +1,153 @@
+releases of a subproject (https://github.com/jupyterlab/jupyterlab/milestones).
+
+* These milestones should be labeled with the version numbers of the releases
+  (`4.3`, `5.0`, etc.).
+* Subprojects should only have a few numbered milestones. Beyond one or two
+  future releases, we have found that detailed milestones become meaningless.
+* Subprojects should also have a `Backlog` milestone for issues and pull
+  requests that have not been assigned to a particular release milestone. This
+  also includes new feature requests ("wishlist") issues that we haven't
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
+* Subprojects should have a `No Action` milestone to indicate issues or pull
+  requests that were closed with no action or work being performed. In general
+  we try to close issues/PRs with this milestone.
+* The only time an issue should have no milestone is early on in the triage
+  process when it still has the `status:Needs Info` label.
+
+## Labels
+
+We use GitHub labels to organize issues and pull requests
+(https://github.com/jupyterlab/jupyterlab/labels). The Github issues namespace
+is completely unstructured; each label is simply a name with an associated
+color. This unstructured nature, makes it difficult to organize work
+effectively. To address, this challenge, we are started to use a more formal
+label schema/taxonomy.
+
+First, every label should have one of four prefixes:
+
+1. `type:`
+2. `component:`
+3. `cat:`
+4. `status:`
+
+All issues of a given prefix should have the same color to make visual
+identification easy.
+
+### `type:`
+
+The `type:` prefix indicates the type of issue or pull request. There are four
+types:
+
+* `type:bug`
+* `type:enhancement`
+* `type:maintenance`
+* `type:question`
+
+Guidelines:
+
+* Each issue or pull request should have only one `type:` prefixed label.
+* Limiting ourselves to these four types allows us to query our issues in a 
+  structural and relational manner.
+
+### `component:`
+
+The `component:` prefix indicates the component or part of the software the
+issue is related to. Thus, on the JupyterLab repository, we have labels such as:
+
+* `component:About`
+* `component:Application`
+* `component:Build System`
+* `component:Command Palette`
+* `component:Completer`
+* `component:Console`
+* `component:Notebook`
+* `component:Terminal`
+* `component:File Browser`
+* `component:File Editor`
+* ...
+
+Guidelines:
+
+* A single issue or pull request can have multiple `component:` labels, but
+  usually it will be a small number.
+* Each subproject will have its own unique set of `component:` labels.
+
+### `cat:`
+
+The `cat:` prefix indicates the "category" of the issue. The category is sort of
+mixin to the `component:` label to indicate cross-cutting categories such as:
+
+* `cat:Accesibility`
+* `cat:Design and UX`
+* `cat:Documentation`
+* `cat:Examples`
+* `cat:New Contributor`
```

Comment:
FYI Github's  new convention would suggest that this line should be changed to `good first issue` See #40